test/Microsoft.TemplateEngine.Mocks/MockCreationEffects.cs

```diff
@@ -0,0 +1,83 @@
+using Microsoft.TemplateEngine.Abstractions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Xunit.Abstractions;
+
+namespace Microsoft.TemplateEngine.Mocks
+{
+    public class MockCreationEffects : ICreationEffects, ICreationEffects2, IXunitSerializable
+    {
+        public MockCreationEffects()
+        {
+
+        }
+
+        public MockCreationEffects WithPrimaryOutputs(params string[] primaryOutputs)
+        {
+            _primaryOutputs = _primaryOutputs.Concat(primaryOutputs).ToArray();
```

Comment:
Since we need resizing on this, maybe a `List` would be better rather than an `Array`.

Reply:
We can use `List`, however `List` is not supported by serializer. We would need to call `ToArray` / `ToList` in serialize/deserialize than. Not sure if we will gain anything.